pkg/metadata/host/host.go

```diff
@@ -34,13 +34,28 @@ func GetPayload(hostname string) *Payload {
 	meta := getMeta()
 	meta.Hostname = hostname
 
-	return &Payload{
+	p := &Payload{
 		Os:            runtime.GOOS,
 		PythonVersion: getPythonVersion(),
 		SystemStats:   getSystemStats(),
 		Meta:          meta,
 		HostTags:      getHostTags(),
 	}
+
+	// Cache the metadata for use in other payload
+	key := buildKey("payload")
+	cache.Cache.Set(key, p, cache.NoExpiration)
+
+	return p
+}
+
+// GetPayloadFromCache returns the payload from the cache if it exists, otherwise it creates it.
+func GetPayloadFromCache(hostname string) *Payload {
```

Comment:
why another function? I'd do this transparently in `GetPayload`

Reply:
I figured when we sent metadata we'd want it fresh

Reply:
Gotcha, let's keep it like this